### PR TITLE
Sidebar

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/yehm9tvrj6eyj2fhpauvdq6jur

[download apk](http://filestorage.bluewallet.io:1488/1598765502.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/72xfth0468d5j2tqyarqr0my4m

[download apk](http://filestorage.bluewallet.io:1488/1598768077.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/7mf1fbeyxew18rc8vx2h509a30

[download apk](http://filestorage.bluewallet.io:1488/1598771237.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/qw1v459amu8yczpnup33p19p60

[download apk](http://filestorage.bluewallet.io:1488/1598804578.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/vrp9rtbw15qajzbwmfv1g2ttyr

[download apk](http://filestorage.bluewallet.io:1488/1598835100.apk)

**Review by Overtorment — CHANGES_REQUESTED** [review]
deepscan & ios build failed.
didnt look at the code

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/patgzfppaxc2yzcqbbe5wmc0wr

[download apk](http://filestorage.bluewallet.io:1488/1598914653.apk)

**Review by Overtorment — CHANGES_REQUESTED** [review]
conflicts. 16 commits. requires tidying up

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/n3z6e4d7cp55xjmnb9kbtmfjem

[download apk](http://filestorage.bluewallet.io:1488/1598964798.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/q5wpft927f8ktd1xmwyxfc6fyg

[download apk](http://filestorage.bluewallet.io:1488/1598969417.apk)

**Comment by marcosrdz** [comment]
> conflicts. 16 commits. requires tidying up

conflict fixes. squashed.

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/2cwyec3dn4gz5f19354maxhua8

[download apk](http://filestorage.bluewallet.io:1488/1598971038.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/3xdpaugc9yjr83g0zxc28y4mpw

[download apk](http://filestorage.bluewallet.io:1488/1598973999.apk)

File: android/app/src/main/res/values-sw600dp/bools.xml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <bool name="portrait_only">false</bool>
```
Comment (Overtorment) [inline android/app/src/main/res/values-sw600dp/bools.xml:3]: wtf is dis

File: screen/wallets/drawerList.js [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+import * as NavigationService from '../../NavigationService';
+import loc from '../../loc';
+import { BlueCurrentTheme } from '../../components/themes';
+import { useTheme, useFocusEffect } from '@react-navigation/native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+const EV = require('../../blue_modules/events');
+const BlueApp: AppStorage = require('../../BlueApp');
+const BlueElectrum = require('../../blue_modules/BlueElectrum');
+
+const DrawerList = props => {
```
Comment (Overtorment) [inline screen/wallets/drawerList.js:19]: did you alter any refresh logic here?

File: screen/wallets/list.js [hunk 3]
```diff
@@ -144,6 +129,7 @@ export default class WalletsList extends Component {
       scrollToEnd = wallets.length > this.state.wallets.length;
     }
 
+    BlueApp.getTransactions(null, 10);
```
Comment (Overtorment) [inline screen/wallets/list.js:132]: wtf

File: screen/wallets/drawerList.js [hunk 4]
```diff
@@ -0,0 +1,311 @@
...
+        console.warn(err);
+      }
+      if (noErr) await BlueApp.saveToDisk(); // caching
+
+      redrawScreen();
+    });
+  };
+
+  const redrawScreen = (scrollToEnd = false) => {
+    console.log('wallets/list redrawScreen()');
```
Comment (Overtorment) [inline screen/wallets/drawerList.js:56]: not up-to-date console log

File: screen/wallets/drawerList.js [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+
+    // here, when we receive REMOTE_TRANSACTIONS_COUNT_CHANGED we fetch TXs and balance for current wallet.
+    // placing event subscription here so it gets exclusively re-subscribed more often. otherwise we would
+    // have to unsubscribe on unmount and resubscribe again on mount.
```
Comment (Overtorment) [inline screen/wallets/drawerList.js:60]: so whats the navigation stack here?
we used to have: first walletsList opened, and then walletTransactions opened, and always one screen was displayed, the other one - not, so the one who is being displayed is exclusively subscribes to REMOTE_TRANSACTIONS_COUNT_CHANGED event and fetches transactions for current wallet if its fired.

and now? wheres the drawerList in all this..?

File: screen/wallets/drawerList.js [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+    EV(EV.enum.TRANSACTIONS_COUNT_CHANGED);
+    if (scrollToEnd) {
+      // eslint-disable-next-line no-unused-expressions
+      walletsCarousel.current?.snapToItem(wallets.length - 2);
+    }
+  };
+
+  useEffect(() => {
+    EV(EV.enum.WALLETS_COUNT_CHANGED, () => redrawScreen(true));
+    console.log('wallets/list componentDidMount');
```
Comment (Overtorment) [inline screen/wallets/drawerList.js:78]: not up-to-date console log

**Review by Overtorment — CHANGES_REQUESTED** [review]
found some bugs on appetize

1)
image

2) wallet cards are not updated when wallet is created or deleted.
STR: open appetizer. choose nexus 9; choose android 10; click arrow to tilt the device. open android's OS drawer and click 'auto-rotate'. create a wallet - wallet is created, but there is no wallet's card on the left:


image

**Comment by Overtorment** [comment]
@ncoelho you approve this layout for tablet/desktop?

File: android/app/src/main/res/values-sw600dp/bools.xml (hunk 1 above)
Comment (marcosrdz) [reply android/app/src/main/res/values-sw600dp/bools.xml:3]: its how you determine on android if the device's resolution is enough to handle splitview or not

File: screen/wallets/drawerList.js (hunk 2 above)
Comment (marcosrdz) [reply screen/wallets/drawerList.js:19]: no pull to refresh on drawerlist. thats on lists.js

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/yywgbj96zpjf5xeeyr5n2ze2qw

[download apk](http://filestorage.bluewallet.io:1488/1599051558.apk)

**Comment by ncoelho** [comment]
Layout wise yes, would like to test it on macOS first

**Comment by ncoelho** [comment]
Problems I found:

Lightning receive screen, CTA is unaligned 
Screenshot 2020-09-02 at 16 12 10

Scan on lightning is giving a blank/black screen
Screenshot 2020-09-02 at 16 12 28

Onchain send screen is algo misaligned, I suspect it is related with batching
Screenshot 2020-09-02 at 16 13 29

If I have a lightning wallet open and tap on the onchain wallet, it doesn't open:
Screenshot 2020-09-02 at 16 13 03

The sidebar is a bit small, I will give a 320px size
Screenshot 2020-09-02 at 16 17 15

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/8kfzdqyag06461cedkc89hkmec

[download apk](http://filestorage.bluewallet.io:1488/1599092125.apk)

**Comment by marcosrdz** [comment]
> Problems I found:
> 
> 
> 
> Lightning receive screen, CTA is unaligned 
> 
> Screenshot 2020-09-02 at 16 12 10
> 
> 
> 
> Scan on lightning is giving a blank/black screen
> 
> Screenshot 2020-09-02 at 16 12 28
> 
> 
> 
> Onchain send screen is algo misaligned, I suspect it is related with batching
> 
> Screenshot 2020-09-02 at 16 13 29
> 
> 
> 
> If I have a lightning wallet open and tap on the onchain wallet, it doesn't open:
> 
> Screenshot 2020-09-02 at 16 13 03
> 
> 
> 
> The sidebar is a bit small, I will give a 320px size
> 
> Screenshot 2020-09-02 at 16 17 15
> 
> 

Check with latest DMG. 320 is the default width

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/cr1566vxu76qm2gkbphymf2fr4

[download apk](http://filestorage.bluewallet.io:1488/1599101969.apk)

**Comment by ncoelho** [comment]
Bugs not found on latest build 👍

**Review by ncoelho — APPROVED** [review]

**Review by limpbrains — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/35yptdz1xq707tczf2fyxyfejg

[download apk](http://filestorage.bluewallet.io:1488/1599181345.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/3u2bbp2jder859xgjrz5x5jzrw

[download apk](http://filestorage.bluewallet.io:1488/1599185689.apk)

File: screen/wallets/drawerList.js (hunk 5 above)
Comment (marcosrdz) [reply screen/wallets/drawerList.js]: remote tx count changed listeners is now in drawerList. All the other ones are listening to the tx count

**Comment by marcosrdz** [comment]
@Overtorment  re-review when you get the chance

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/2zkqzetw23e0updc8473hmrhdr

[download apk](http://filestorage.bluewallet.io:1488/1599270325.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/6g496p3qepe53nkvk8j2rxafnw

[download apk](http://filestorage.bluewallet.io:1488/1599447648.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/4qxg41abpb549twbcbzndevtcg

[download apk](http://filestorage.bluewallet.io:1488/1599495846.apk)

**Review by Overtorment — CHANGES_REQUESTED** [review]
can you elaborate more how drawerList works? whats the navigation stack? who calls whom? who being inside whom?

also, pls fix conflicts

File: Navigation.js [hunk 7]
```diff
@@ -1,7 +1,8 @@
 // import { createAppContainer } from '@react-navigation/native';
```
[the remark this answers is not in the document]
Comment (marcosrdz) [reply Navigation.js]: Most likely accidental when doing autocomplete.

**Comment by marcosrdz** [comment]
> can you elaborate more how drawerList works? whats the navigation stack? who calls whom? who being inside whom?
> 
> also, pls fix conflicts

DrawerList is this: https://reactnavigation.org/docs/drawer-based-navigation

we assign it as 'permanent' so it behaves like a split view controller we see on iPadOS. It will always be mounted. it's width will be determined by the device real estate. if it has enough width, it'll be set to the default 320 width, otherwise..its set to 0.
All stacks "behave independently" but are able to communicate to each other via navigate calls.

**Review by Overtorment — APPROVED** [review]
lgtm. I want to test it on my device first, and then lets merge

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/tmvy5vm20yn33b455xxktwymcc

[download apk](http://filestorage.bluewallet.io:1488/1599577221.apk)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/z19xxqvfh42a8q5unpqa3e1pn4

[download apk](http://filestorage.bluewallet.io:1488/1599580876.apk)

**Pull request merged**